packages/design-system/src/components/table/useTable/index.tsx

```diff
@@ -91,48 +95,114 @@ export type PersistentStorageData = {
+    row: TableRow
+  ) => void;
+  getRowObjectKey: (row: TableRow) => string;
+  conditionalTableRowClassesHandler?: (
+    row: TableRow,
+    isRowFocused: boolean,
+    rowIndex: number
+  ) => string;
+  exportTableData?: (rows: TableRow[]) => void;
+  hasVerticalBar?: (row: TableRow) => boolean;
+}
+
+export interface TableStoreContext {
+  state: {
+    columns: TableColumn[];
+    hideableColumns: TableColumn[];
+    rows: TableRow[];
+    sortKey: ColumnSortingOutput['sortKey'];
+    sortOrder: ColumnSortingOutput['sortOrder'];
+    areAllColumnsVisible: ColumnVisibilityOutput['areAllColumnsVisible'];
+    tableContainerRef: ColumnResizingOutput['tableContainerRef'];
+    isResizing: ColumnResizingOutput['isResizing'];
+    filters: TableFilter;
+    selectedFilters: TableFilter;
+    isFiltering: TableFilteringOutput['isFiltering'];
+    searchValue: TableSearchOutput['searchValue'];
+  };
+  actions: {
+    setSortKey: ColumnSortingOutput['setSortKey'];
+    setSortOrder: ColumnSortingOutput['setSortOrder'];
+    hideColumn: ColumnVisibilityOutput['hideColumn'];
+    toggleVisibility: ColumnVisibilityOutput['toggleVisibility'];
+    showColumn: ColumnVisibilityOutput['showColumn'];
+    isColumnHidden: ColumnVisibilityOutput['isColumnHidden'];
+    onMouseDown: ColumnResizingOutput['onMouseDown'];
+    toggleFilterSelection: TableFilteringOutput['toggleFilterSelection'];
+    toggleSelectAllFilter: TableFilteringOutput['toggleSelectAllFilter'];
+    resetFilters: TableFilteringOutput['resetFilters'];
+    isSelectAllFilterSelected: TableFilteringOutput['isSelectAllFilterSelected'];
+    setSearchValue: TableSearchOutput['setSearchValue'];
+    onRowClick: useTableProps['onRowClick'];
+    onRowContextMenu: useTableProps['onRowContextMenu'];
+    getRowObjectKey: useTableProps['getRowObjectKey'];
+    conditionalTableRowClassesHandler?: useTableProps['conditionalTableRowClassesHandler'];
+    exportTableData?: useTableProps['exportTableData'];
+    hasVerticalBar?: useTableProps['hasVerticalBar'];
+  };
 }
 
-const useTable = ({
+const initialState: TableStoreContext = {
+  state: {
+    columns: [],
+    hideableColumns: [],
+    rows: [],
+    sortKey: '',
+    sortOrder: 'asc',
+    areAllColumnsVisible: true,
+    tableContainerRef: null,
+    isResizing: false,
+    filters: {},
+    selectedFilters: {},
+    isFiltering: false,
+    searchValue: '',
+  },
+  actions: {
+    setSortKey: noop,
+    setSortOrder: noop,
+    hideColumn: noop,
+    toggleVisibility: noop,
+    showColumn: noop,
+    isColumnHidden: () => false,
+    onMouseDown: noop,
+    toggleFilterSelection: noop,
+    toggleSelectAllFilter: noop,
+    resetFilters: noop,
+    isSelectAllFilterSelected: () => false,
+    setSearchValue: noop,
+    onRowClick: noop,
+    onRowContextMenu: noop,
+    getRowObjectKey: () => '',
+  },
+};
+
+export const TableContext = createContext<TableStoreContext>(initialState);
+
+export const TableProvider = ({
```

Comment:
@mayan-000 can we rename the folder as tableProvider and can we split this into separate parts like `useTable.ts`, `provider.tsx`, `context.ts`?